mixed-format/loader.mjs

```diff
@@ -0,0 +1,34 @@
+export async function load(url, context, nextLoad) {
+  if (!url.endsWith('.js')) return nextLoad(url);
+
+  const nextResult = await nextLoad(url, { ...context, format: 'module' })
+    .then((result) => {
+      if (containsCJS(result.source)) { throw new Error('CommonJS'); }
+
+      return result;
+    })
+    .catch(async (err) => {
+      if (
+        (err?.message.includes('require') && err.includes('import'))
+        || err?.message.includes('CommonJS')
+      ) {
+        return { format: 'commonjs' };
+      }
+
+      throw err;
+    });
+
+  return nextResult;
+}
+
+function containsCJS(source) {
+  const src = '' + source;
+
+  // A realistic version of this loader would use a parser like Acorn to check for actual `module.exports` syntax
+  if (src.match(/exports[\.( ?=)]/)) { return true };
+
+  if (
+    src.match(/require\(/)
+    && !src.match(/createRequire\(/)
+  ) return true;
```

Comment:
We should be using `/regex/.test` here, much faster than `'string'.match` as it doesn't need to create an array.

Reply:
Ahh, good call. I'll send an update this evening